core/src/software/aws/toolkits/core/utils/S3ClientUtils.kt

```diff
@@ -0,0 +1,12 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.core.utils
+
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.Bucket
+import software.aws.toolkits.core.s3.regionForBucket
+
+fun S3Client.listBucketsByRegion(filterRegionId: String?): Sequence<Bucket> = this.listBuckets().buckets()
```

Comment:
```kotlin
    fun S3Client.listBucketsByRegion(filterRegionId: String?): Sequence<Bucket> = this.listBuckets().buckets()
        .asSequence()
        .also { buckets ->
            filterRegionId?.let { _ ->
                buckets.filter { this.regionForBucket(it.name()) == filterRegionId }
            }
        }

Same functionality but doesn't pay the cost of the additional network call if the region id is null
```

Reply:
nit: I'd say that this function should not accept `null` - if you're trying to filter by a region it's the callers responsibility to ensure they have a region.

nit <span>#</span>2: `filterRegionId` could be renamed `regionId` - it's already obvious from the method name what its purpose is

Reply:
For Nit 1, this was deliberate, so that you could hook this to more data driven sources, like a separate UI control that lists regions. But this makes me think that selecting "no region" would show all buckets, which is likely incorrect. Fixing this one.

Fixing Nit 2